mlir/lib/Conversion/AMDGPUToROCDL/AMDGPUToROCDL.cpp

```diff
@@ -76,11 +83,168 @@ static Value getLinearIndexI32(ConversionPatternRewriter &rewriter,
   return index ? index : createI32Constant(rewriter, loc, 0);
 }
 
+/// Compute the contents of the `num_records` field for a given memref
+/// descriptor - that is, the number of bytes that's one element past the
+/// greatest possible valid index into the memref.
+static Value getNumRecords(ConversionPatternRewriter &rewriter, Location loc,
+                           MemRefType memrefType,
+                           MemRefDescriptor &memrefDescriptor,
+                           ArrayRef<int64_t> strides,
+                           uint32_t elementByteWidth) {
+  if (memrefType.hasStaticShape() &&
```

Comment:
I think we can have an assert before 
```suggestion
assert(memrefType.hasStaticShape() && ...)
  ```

Reply:
... This isn't an assert

This is an `if (case) { ...; return vCase;} ...`